src/rime/context.cc

```diff
@@ -128,10 +128,15 @@ bool Context::Highlight(size_t index) {
     return false;
   Segment& seg(composition_.back());
   size_t new_index = index;
-  size_t candidate_count = seg.menu->Prepare(index + 1);
-  if (index >= candidate_count) {
-    DLOG(INFO) << "selection index exceeds candidate pool, fallback to last";
-    new_index = candidate_count - 1;
+  if (index < 0) {
+    DLOG(INFO) << "selection index < 0, fallback to 0";
+    new_index = 0;
+  } else {
+    size_t candidate_count = seg.menu->Prepare(index + 1);
+    if (index >= candidate_count) {
+      DLOG(INFO) << "selection index exceed candidate pool, fallback to last";
+      new_index = candidate_count - 1;
```

Comment:
If index==0 and candidate_count==0, new_index = -1; still got invalid index.

If logs are not required this should work:
`(std::max)(0, (std::min)(candidate_count - 1, index))`

Reply:
I pushed new changes to this branch